### PR TITLE
Update actions running on deprecated Node v16

### PR DESCRIPTION
**Pull request by dhruvkb**
## Description

This PR updates the actions that are still running on Node 16 and raising deprecation warnings. They have been updated to their latest versions.

In the case of now-unmaintained action `extractions/setup-just`, I have switched to a different maintained one.

One action `peaceiris/actions-gh-pages` has not been updated to Node v20 yet (PR https://github.com/peaceiris/actions-gh-pages/pull/1067 merged, no release since). There is a possibility to drop it altogether, in favour of official actions https://github.com/actions/upload-pages-artifact and https://github.com/actions/deploy-pages but those that can be a separate issue.

## Testing Instructions

0. Check that CI continues to run. 
1. Refer to the breaking changes for the action from their docs and ensure we're not affected by them.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: .github/actions/setup-env/action.yml [hunk 1]
```diff
@@ -19,14 +19,14 @@ runs:
   steps:
     # Just
     - name: Setup just
-      uses: extractions/setup-just@v1
-      env:
-        GITHUB_TOKEN: ${{ github.token }}
+      uses: taiki-e/install-action@v2
```
Comment (obulat) [inline .github/actions/setup-env/action.yml:22]: Great find!

File: .github/actions/setup-env/action.yml [hunk 2]
```diff
@@ -46,12 +46,12 @@ runs:
     # Node.js
     - name: Setup pnpm
       if: inputs.setup_nodejs == 'true'
-      uses: pnpm/action-setup@v2
+      uses: pnpm/action-setup@v3
```
Comment (obulat) [inline .github/actions/setup-env/action.yml:49]: Not blocking for this PR, but do we really need a separate action for pnpm setup? Could we replace it with `corepack enable`, like it's done in, e.g., elk: https://github.com/elk-zone/elk/blob/main/.github/workflows/ci.yml

**Review by obulat — APPROVED** [review]
Thank you for finding all of the updatable actions!

**Review by krysal — APPROVED** [review]
LGTM

File: .github/actions/setup-env/action.yml (hunk 2 above)
Comment (dhruvkb) [reply .github/actions/setup-env/action.yml:49]: Good point, corepack seems to work quite well for me locally so we can explore it in the action as well, but I would prefer to do that in a separate issue and PR.

**Pull request merged**